src/handlers/profile/attributes.rs

```diff
@@ -0,0 +1,180 @@
+use {
+    super::{
+        super::HANDLER_TASK_METRICS,
+        utils::{check_attributes, is_timestamp_within_interval},
+        RegisterRequest,
+        UpdateAttributesPayload,
+        UNIXTIMESTAMP_SYNC_THRESHOLD,
+    },
+    crate::{
+        database::helpers::{get_name_and_addresses_by_name, update_name_attributes},
+        error::RpcError,
+        state::AppState,
+        utils::crypto::{constant_time_eq, verify_message_signature},
+    },
+    axum::{
+        body::Bytes,
+        extract::{Path, State},
+        response::{IntoResponse, Response},
+        Json,
+    },
+    hyper::StatusCode,
+    sqlx::Error as SqlxError,
+    std::{str::FromStr, sync::Arc},
+    tracing::log::{error, info},
+    wc::future::FutureExt,
+};
+
+pub async fn handler(
+    state: State<Arc<AppState>>,
+    name: Path<String>,
+    body: Bytes,
+) -> Result<Response, RpcError> {
+    handler_internal(state, name, body)
+        .with_metrics(HANDLER_TASK_METRICS.with_name("profile_attributes_update"))
+        .await
+}
+
+#[tracing::instrument(skip(state))]
+pub async fn handler_internal(
+    state: State<Arc<AppState>>,
+    Path(name): Path<String>,
+    body: Bytes,
```

Comment:
Would this be simpler?
```suggestion
    Json(request_payload): Json<RegisterRequest>
```

Reply:
Didn't know that. Changed it here and in the registration handler. Thanks.